src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs

```diff
@@ -12,6 +12,7 @@ public class StringReader : TextReader
         private string? _s;
         private int _pos;
         private int _length;
+        private CachedCompletedTask<int> _lastReadTask;
```

Comment:
could you please provide a benchmark that shows that it's beneficial to cache the task here?

I would expect all the users who care about perf to use the `ValueTask`-based overloads.

Reply:
Good point, optimizing the `Task`-returning overloads is not worth the hassle. I reverted changes to this file.